packages/server/src/automations/automationUtils.ts

```diff
@@ -122,17 +122,88 @@ export function getError(err: any) {
 }
 
 export function guardAttachment(attachmentObject: any) {
-  if (
-    attachmentObject &&
-    (!("url" in attachmentObject) || !("filename" in attachmentObject))
-  ) {
+  if (!attachmentObject) {
+    return
+  }
+  if (typeof attachmentObject !== "object") {
+    throw new Error(
+      `Attachments must be objects with both "url" and "filename" keys.`
+    )
+  }
+  if (!("url" in attachmentObject) || !("filename" in attachmentObject)) {
     const providedKeys = Object.keys(attachmentObject).join(", ")
     throw new Error(
       `Attachments must have both "url" and "filename" keys. You have provided: ${providedKeys}`
     )
   }
 }
 
+function deriveFilenameFromUrl(url?: string) {
+  if (!url) return ""
+  try {
+    const pathname = url.startsWith("http") ? new URL(url).pathname : url
+    const parts = pathname.split("/")
+    return parts[parts.length - 1] || ""
+  } catch {
+    return ""
+  }
+}
+
+function normalizeSingleAttachment(input: any): AutomationAttachment | null {
```

Comment:
Use unknown for this scenario. Unknown type forces the usage of the `typeof` checks which you have done below.